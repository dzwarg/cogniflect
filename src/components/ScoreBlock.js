import React from 'react'
import {Col} from 'react-bootstrap'

const fmt = (val) => (`${(100 * val).toFixed(2)} %`)
const ScoreBlock = ({type, correct, total}) => (
  <Col xs={12} md={6}>
    <h2>{type}</h2>
    <h3>Score: {fmt(correct/total)}</h3>
    <p>You answered {correct} out of {total} correctly.</p>
  </Col>
)

export default ScoreBlock