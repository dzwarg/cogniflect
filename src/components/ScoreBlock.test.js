import ScoreBlock from './ScoreBlock'

const initialState = {
  type: 'Quasimodally',
  correct: 4,
  total: 8
}

describe('components/ScoreBlock', () => {
  it('matches snapshot', () => {
    const component = ScoreBlock(initialState)
    expect(component).toMatchSnapshot()
  })
})