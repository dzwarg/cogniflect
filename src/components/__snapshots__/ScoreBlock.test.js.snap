// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ScoreBlock matches snapshot 1`] = `
<ForwardRef(Bootstrap(Col))
  md={6}
  xs={12}
>
  <h2>
    Quasimodally
  </h2>
  <h3>
    Score: 
    50.00 %
  </h3>
  <p>
    You answered 
    4
     out of 
    8
     correctly.
  </p>
</ForwardRef(Bootstrap(Col))>
`;
